src/Illuminate/Routing/Router.php

```diff
@@ -1309,6 +1310,45 @@ public function setContainer(Container $container)
         return $this;
     }
 
+    /**
+     * Discover and routes resource controllers by the given path.
+     *
+     * @param  string  $path
+     * @return void
+     */
+    public function discover($path)
+    {
+        $files = (new Finder)->in($path)->files();
```

Comment:
This will remove the need to check the filename in line 1340:

```php
$files = (new Finder)->in($path)->files()->notName('Controller.php');
```

Reply:
This is a POC.